docs/lab-hooks.md

```diff
@@ -0,0 +1,352 @@
+
+There are two settings that you can change in the `assessmentVariables` function that will affect the behavior of the `modifySubmissionScores` function:
+
+- `previous_submissions_lookback`: The number of previous submissions to look back when calculating the score. By default, it is set to 1000.
+- `exclude_autograding_in_progress_submissions`: If set to `true`, the submissions that are currently being autograded will be excluded when passed into the `modifySubmissionScores` function. By default, it is set to `false`.
+
+The three arguments passed into the `modifySubmissionScores` function are:
+
+- `scores`: A hash that maps the problem name to the score.
+- `previous_submissions`: A list of previous submissions, sorted by submission time in descending order, it is an ActiveRecord object.
+- `problems`: A list of problems in the lab, it is an ActiveRecord object.
+
+For more information on how to use ActiveRecord, please refer to the [ActiveRecord documentation](http://guides.rubyonrails.org/active_record_querying.html). For the schema of the `Submission` and `Problem` models, please refer to the [Autolab Schema](https://github.com/autolab/Autolab/blob/master/db/schema.rb).
+
+## Raw Score Calculations
+
+Hook: `raw_score`
+
+By default, the raw score for a submission is the sum of the individual problem scores. This hook allows you to override the raw score calculation for a lab.
+
+```ruby
+def raw_score(score)
+    perfindex = score["Autograded Score"].to_f()
+    heap = score["Heap Checker"].to_f()
+    style = score["Style"].to_f()
+    deduct = score["CorrectnessDeductions"].to_f()
+    perfpoints = perfindex
+
+    # perfindex below 50 gets autograded score of 0.
+    if perfindex < 50.0 then
+        perfpoints = 0
+    else
+        perfpoints = perfindex
+    end
+
+    return perfpoints + heap + style + deduct
+end
+```
+
+This particular lab has four problems called "Autograded Score", "Heap Checker", "Style", and "CorrectnessDeductions". The code snippet above sets an "Autograded Score" of less than 50 to zero when the raw score is calculated.
+
+## Submission File MIME Type Check
+
+Hook: `checkMimeType`
+
+By default, Autolab does not perform MIME type check for submission files. This hook allows you to define a MIME type check method. For example, to prevent students from submitting a binary file to the assessment, you might add the following `checkMimeType` function:
+
+```ruby
+def checkMimeType(contentType, fileName)
+    return contentType != "application/octet-stream"
+end
+```
+
+As of now, the only way to provide a more informative message to student is to raise an error:
```

Comment:
Nit: the only way to provide a more informative message to a student ...

Reply:
Missing the 'a'

Reply:
Resolved, by replacing "student" with "students". Also fixed a related issue below "to do file type check[s]"